### DISCUSSION TITLE
How does prisma generate CUIDs on the db?

### DISCUSSION THREAD

**Discussion by special-character** [prisma/prisma#2914] [Q&A] [answered]
I am writing a SQL script and want to generate a cuid in SQL. I feel like prisma might do this in SQL but I didn't see it in the source on first look.

Is there a way to generate cuid in sql? I saw in the docs that the query engine is responsible for this so not sure if it is done in SQL or not.

**Comment by ryands17** [comment, marked as answer]
I don't think you might be able to generate a CUID natively in SQL. You could try by writing a script in `.js` or `.ts` and perform SQL operations via `prisma.queryRaw` and `prisma.executeRaw` and use any CUID generator package.

**Reply by pantharshit00** [reply]
We use this package in the engine to generate the cuids: https://github.com/prisma/cuid-rust

Node has also a good package for cuids: https://github.com/ericelliott/cuid

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
